src/device-manager/java/src/nl/Weave/DeviceManager/NetworkInfo.java

```diff
@@ -45,6 +45,7 @@ public NetworkInfo()
         ThreadNetworkName = null;
         ThreadExtendedPANId = null;
         ThreadNetworkKey = null;
+        ThreadPSKc = null;
```

Comment:
A bunch of other code in `Make` and `MakeThread` below will need to be modified to account for a new parameter.

Reply:
The work items listed here remain unaddressed.

Reply:
I stand by my previous comment.  I would expect a few more changes in NetworkInfo.java

Reply:
Please apply the following patch:

From 8ece3473237630f331257f022be6b41379201697 Mon Sep 17 00:00:00 2001
From: Robert Szewczyk <rszewczyk@nestlabs.com>
Date: Fri, 2 Aug 2019 13:54:06 -0700
Subject: [PATCH] Outstanding changes to fully flesh out PSKc support.

---
 .../java/src/nl/Weave/DeviceManager/NetworkInfo.java         | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/device-manager/java/src/nl/Weave/DeviceManager/NetworkInfo.java b/src/device-manager/java/src/nl/Weave/DeviceManager/NetworkInfo.java
index 71de17a6..e157182e 100644
--- a/src/device-manager/java/src/nl/Weave/DeviceManager/NetworkInfo.java
+++ b/src/device-manager/java/src/nl/Weave/DeviceManager/NetworkInfo.java
@@ -151,7 +151,7 @@ public class NetworkInfo
     public static NetworkInfo Make(int networkType, long networkId,
                                    String wifiSSID, int wifiMode, int wifiRole, int wifiSecurityType, byte[] wifiKey,
                                    String threadNetworkName, byte[] threadExtendedPANId, byte[] threadNetworkKey,
-                                   short wirelessSignalStrength, int threadPANId, int threadChannel)
+                                   byte[] threadPSKc, short wirelessSignalStrength, int threadPANId, int threadChannel)
     {
         NetworkInfo netInfo = new NetworkInfo();
         netInfo.NetworkType = nl.Weave.DeviceManager.NetworkType.fromVal(networkType);
@@ -164,6 +164,7 @@ public class NetworkInfo
         netInfo.ThreadNetworkName = threadNetworkName;
         netInfo.ThreadExtendedPANId = threadExtendedPANId;
         netInfo.ThreadNetworkKey = threadNetworkKey;
+        netInfo.ThreadPSKc = threadPSKc;
         netInfo.ThreadPANId = threadPANId;
         netInfo.ThreadChannel = threadChannel;
         netInfo.WirelessSignalStrength = wirelessSignalStrength;
@@ -176,7 +177,7 @@ public class NetworkInfo
                                    short wirelessSignalStrength)
     {
         return Make(networkType, networkId, wifiSSID, wifiMode, wifiRole, wifiSecurityType, wifiKey, threadNetworkName,
-                    threadExtendedPANId, threadNetworkKey, wirelessSignalStrength, THREAD_PANID_NOTSPECIFIED,
+                    threadExtendedPANId, threadNetworkKey, null, wirelessSignalStrength, THREAD_PANID_NOTSPECIFIED,
                     THREAD_CHANNEL_NOTSPECIFIED);
     }
 }
-- 
2.22.0.770.g0f2c4a37fd-goog

Reply:
The above patch should preferably be squashed into one of your commits.  I believe that this is the last outstanding patch required to merge; I will approve and merge this PR after your update

Reply:
Updated